control-plane/api/v1alpha1/servicedefaults_types.go

```diff
@@ -396,6 +408,7 @@ func (in *Upstream) toConsul() *capi.UpstreamConfig {
 		Name:               in.Name,
 		Namespace:          in.Namespace,
 		Partition:          in.Partition,
+		Peer:               in.Peer,
```

Comment:
I think we need an additional validation check that both Peer and Partition are not provided simultaneously IIRC?

Reply:
Yep, that's a good callout.